improver/psychrometric_calculations/hail_size.py

```diff
@@ -0,0 +1,295 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""modules to calculate hail_size"""
+
+import math
+from operator import itemgetter
+from typing import List
+
+import iris
+import numpy as np
+from iris.cube import Cube
+
+from improver import BasePlugin
+from improver.metadata.utilities import (
+    create_new_diagnostic_cube,
+    generate_mandatory_attributes,
+)
+from improver.psychrometric_calculations.psychrometric_calculations import (
+    adjust_for_latent_heat,
+    dry_adiabatic_temperature,
+    saturated_humidity,
+)
+from improver.utilities.cube_manipulation import MergeCubes
+
+
+class HailSize(BasePlugin):
+    def __init__(self, model_id_attr: str = None):
+        self.model_id_attr = model_id_attr
+        self.ccl_temperature = None
+        self.ccl_pressure = None
+
+    def nomogram_values(self):
+
+        """set-up dictionary of the table found in https://doi.org/10.1002/met.236 of hail sizes"""
+
+        """formatting turned off for this graph as the formatting makes it unreadable. Might need to 
+        reconsider how I'm formatting this"""
+
+        # fmt: off
+        lookup_nomogram={
+            5:  [0 ,0 ,0 ,0 ,0 ,0 ,0 ,2 ,2 ,2 ,2 ,2 ,5 ,5 ,5 ,5 ,10,10,10,10 ,10 ,10 ,10 ,15 ,15 ,15 ],
+            10: [0 ,0 ,0 ,0 ,0 ,2 ,2 ,2 ,5 ,5 ,5 ,10,10,15,15,15,15,15,15,20 ,20 ,20 ,20 ,25 ,25 ,25 ],
+            15: [0 ,0 ,2 ,2 ,2 ,2 ,5 ,10,10,15,15,20,20,20,25,25,25,25,30,30 ,30 ,35 ,35 ,35 ,35 ,35 ],
+            20: [2 ,2 ,2 ,2 ,2 ,5 ,10,15,20,20,20,25,25,30,30,35,35,40,40,40 ,40 ,45 ,45 ,45 ,45 ,50 ],
+            25: [2 ,5 ,5 ,5 ,10,15,20,20,25,30,30,35,40,40,40,40,45,45,45,50 ,50 ,50 ,50 ,50 ,55 ,55 ],
+            30: [5 ,5 ,5 ,10,15,20,20,25,30,35,40,40,40,45,45,50,50,55,55,60 ,60 ,60 ,60 ,65 ,65 ,65 ],
+            35: [5 ,5 ,10,15,20,20,25,30,35,40,40,45,45,50,55,55,60,60,65,70 ,70 ,75 ,75 ,80 ,80 ,80 ],
+            40: [5 ,5 ,10,15,20,20,25,30,35,40,40,45,50,55,60,60,65,70,75,80 ,80 ,85 ,90 ,100,100,100],
+            45: [5 ,10,15,20,20,25,30,35,40,45,45,50,55,60,60,65,70,80,85,90 ,95 ,100,105,110,110,110],
+            50: [5 ,10,15,20,20,25,30,35,40,45,50,50,55,60,65,75,80,85,90,100,105,110,115,120,120,120]  
+        }
```

Comment:
Spaces between values and the subsequent comma is very strange in python, though I see in this case that it makes it nice to read. Black obviously wants to do ridiculous things to this table, but there is probably a happy medium somewhere which is more conformant and not endlessly long and unreadable.